test/Tests.hs

```diff
@@ -0,0 +1,34 @@
+import Test.Hspec
+import Test.QuickCheck
+import Control.Exception (evaluate)
+import BinaryGap
+import OddOccurrencesInArray
+import Data.Bits
+
+main :: IO ()
+main = hspec $ do
+  describe "binaryGap" $ do
+    it "of    7 equals to 0" $ do
+      binaryGapInt 7 `shouldBe` 0
+      
+    it "of   15 equals to 0" $ do
+      binaryGapInt 15 `shouldBe` 0
+      
+    it "of    9 equals to 2" $ do
+      binaryGapInt 9 `shouldBe` 2
+      
+    it "of  529 equals to 4" $ do
+      binaryGapInteger 529 `shouldBe` 4
+      
+    it "of 1041 equals to 5" $ do
+      binaryGap (1041::Integer) `shouldBe` 5
+      
+  describe "OddOccurrencesInArray" $ do
+    it "2" $ do
+      oddOccurrencesInArrayInteger [1, 1, 2] `shouldBe` 2
+
+    it "3" $ do
+      oddOccurrencesInArrayInt [1, 3, 2, 1, 2] `shouldBe` 3
+
+    it "4" $ do
+      oddOccurrencesInArray [3::Integer, 1, 3, 2, 4, 1, 2] `shouldBe` 4
```

Comment:
INFO: I'm curious why you didn't use the specific example listed in the problem above: 
```haskell
-- For example, given array A such that:
--   A[0] = 9  A[1] = 3  A[2] = 9
--   A[3] = 3  A[4] = 9  A[5] = 7
--   A[6] = 9

-- the function should return 7, as explained in the example above.
```

Reply:
Other tests that could be done - longer arrays and larger numbers (close to the maximum value for `Int`, for instance)